org.mapstruct.eclipse/src/org/mapstruct/eclipse/internal/quickfix/fixes/AddIgnoreTargetMappingAnnotationQuickFix.java

```diff
@@ -0,0 +1,207 @@
+import org.eclipse.jdt.core.dom.SingleMemberAnnotation;
+import org.eclipse.jdt.core.dom.StringLiteral;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+import org.eclipse.jdt.core.dom.rewrite.ListRewrite;
+import org.mapstruct.eclipse.internal.quickfix.MapStructQuickFix;
+import org.mapstruct.eclipse.internal.quickfix.visitors.FindAnnotationByNameVisitor;
+
+import static org.mapstruct.eclipse.internal.MapStructAPIConstants.MAPPINGS_FQ_NAME;
+import static org.mapstruct.eclipse.internal.MapStructAPIConstants.MAPPINGS_SIMPLE_NAME;
+import static org.mapstruct.eclipse.internal.MapStructAPIConstants.MAPPING_FQ_NAME;
+import static org.mapstruct.eclipse.internal.MapStructAPIConstants.MAPPING_MEMBER_IGNORE;
+import static org.mapstruct.eclipse.internal.MapStructAPIConstants.MAPPING_MEMBER_TARGET;
+import static org.mapstruct.eclipse.internal.MapStructAPIConstants.MAPPING_SIMPLE_NAME;
+
+/**
+ * Quick fix for error/warning "Unmapped target property" / "Unmapped target properties" that adds
+ * {@code @Mapping( target = "<property>", ignore = true)} to the method.
+ *
+ * @author Andreas Gudian
+ */
+public class AddIgnoreTargetMappingAnnotationQuickFix extends MapStructQuickFix {
+
+    public AddIgnoreTargetMappingAnnotationQuickFix() {
+        super( "Ignore unmapped target properties" );
+    }
+
+    @Override
+    public boolean canFix(IMarker marker) {
+        return getMessage( marker ).startsWith( "Unmapped target propert" );
+    }
+
+    @Override
+    protected ASTRewrite getASTRewrite(CompilationUnit unit, ASTNode nodeWithMarker, IMarker marker) {
+        Collection<String> properties = extractPropertiesFromMessage( getMessage( marker ) );
+
+        if ( properties == null ) {
+            return null;
+        }
+
+        AST ast = unit.getAST();
+
+        ASTRewrite rewrite = ASTRewrite.create( ast );
+        MethodDeclaration method = (MethodDeclaration) nodeWithMarker;
+
+        ListRewrite mappingList = getListForAddingMappingAnnotations( unit, properties, ast, rewrite, method );
+
+        addMappingAnnotations( properties, ast, mappingList );
+
+        addImportIfRequired( unit, rewrite, MAPPING_FQ_NAME );
+
+        return rewrite;
+    }
+
+    private ListRewrite getListForAddingMappingAnnotations(CompilationUnit unit, Collection<String> properties,
+                                                           AST ast, ASTRewrite rewrite, MethodDeclaration method) {
+
+        // if there is already an @Mappings annotation, add the new @Mapping's there
+        Annotation mappingsAnnotation = findAnnotation( method, MAPPINGS_FQ_NAME );
+        if ( mappingsAnnotation != null ) {
+            return rewrite.getListRewrite(
+                ( (SingleMemberAnnotation) mappingsAnnotation ).getValue(),
+                ArrayInitializer.EXPRESSIONS_PROPERTY );
+        }
+
+        // if repeatable @Mapping's are supported, then add the annotations directly to the method
+        if ( supportsRepeatableMapping( unit ) ) {
```

Comment:
Not sure what the lifecycle of those classes/classloaders here is. And it needs to work even if there are projects with and without Java8 in the workspace. Let's leave this as an optimization for later ;-)